File: BL.Objects/List/EventList/EventList.cs
using System;
using System.Collections.Generic;

namespace BL.Objects.List.EventList
{
    public class EventList : BaseList
    {
        public string Title { get; set; }

        public long Public { get; set; }

        public string Description { get; set; }

        public List<ProductList.ProductList> ProductList { get; set; }
    }
}

File: Web/app/objects/list/eventList/eventList.ts
import { ProductList } from "../productList/productList";
import { BaseList } from "../baseList";
/**
 * Created by vadim.m on 11/17/2017.
 */

export class EventList extends BaseList {
    public Description: string;
    public Public: number;
    public ProductList: ProductList[] = new Array<ProductList>();

    public constructor(init?: Partial<EventList>) {
        super();
        Object.assign(this, init);
    }
}